src/Propel/Generator/Platform/MysqlPlatform.php

```diff
@@ -952,6 +952,42 @@ public function doQuoting($text)
         return '`' . strtr($text, ['.' => '`.`']) . '`';
     }
 
+    /**
+     * Get the default On Delete behavior for foreign keys when not explicity set.
+     *
+     * @return string
+     */
+    public function getDefaultForeignKeyOnDeleteBehavior(): string
+    {
+        preg_match('/^(\d+)\.(\d+)\.(\d+)-([^-]+)-?(.*)/', $this->con->getAttribute(PDO::ATTR_SERVER_VERSION), $versionComponents);
+        if (isset($versionComponents[4]) && strcasecmp($versionComponents[4], 'mariadb') != 0) {
+            // assume MySQL
+            if ($versionComponents[1] == '8') {
```

Comment:
Several remarks here:
- this code is not doing `getDefaultForeignKeyOnDeleteBehavior` - it should not be in this method
- the server version string is set by the package builder, trying to match it this closely is extremely fragile
- for now, I would assume that MySQL will stick with `NO ACTION` in later versions, not just 8

Please change this to something like:
```php
class MysqlPlatform extends DefaultPlatform
{
    protected $serverVersion;

    protected function getServerVersion(): ?string
    {
        if(!$this->serverVersion && $this->con) {
            $this->serverVersion = $this->con->getAttribute(PDO::ATTR_SERVER_VERSION);
        }

        return $this->serverVersion;
    }

    protected function getMayorVersionNumber(): ?int
    {
        $serverVersion = $this->getServerVersion();
        if (!$serverVersion) {
            return null;
        }
        $dotPos = strpos($serverVersion, '.');
        if ($dotPos === false) {
            return null;
        }

        return (int) substr($serverVersion, 0, $dotPos - 1);
    }

    protected function isMariaDB(): bool
    {
        $serverVersion = $this->getServerVersion() ?? '';

        return (stripos($serverVersion, 'mariadb') !== false);
    }

    public function getDefaultForeignKeyOnDeleteBehavior(): string
    {
        $mayorVersion = $this->getMayorVersion;

        return ($mayorVersion && $mayorVersion >= 8 && !$this->isMariaDB()) : ForeignKey::NOACTION : ForeignKey::RESTRICT;
    }

    public function getDefaultForeignKeyOnUpdateBehavior(): string
    {
        $mayorVersion = $this->getMayorVersion;

        return ($mayorVersion && $mayorVersion >= 8 && !$this->isMariaDB()) : ForeignKey::NOACTION : ForeignKey::RESTRICT;
    }
```

Reply:
Done